### ISSUE TITLE
MEMOIZATION cache does not work with other namespace except 'argo'

### ISSUE THREAD

**Issue by heidongxianhua** [deepmodeling/dflow#719]
## Checklist

 

* [ ] Checked the syntax is legal within dflow.
* [ ] Tested using the latest version.

## Summary

What happened/what you expected to happen?

What version are you running?

## Diagnostics

Paste the smallest python script that reproduces the bug
when I run the examples/test_big_parameter.py, and change the namespace to `test`, the cache can not hit, but it work well when the namespace is 'argo'.
```python

```
with the namespace 'test'
![image](https://github.com/deepmodeling/dflow/assets/22924371/b8b4dcd0-d30d-4380-a80a-dac71cb79a72)

with the namespace 'argo'
![image](https://github.com/deepmodeling/dflow/assets/22924371/665d9159-0fcf-4ab7-a20a-2c35a10e5c55)

**Comment by saltball** [comment]
Could you please provide details on how you change the namespace? Do you change the `config['namespace']` from `dflow.config`, or do you use `Workflow(...,namespace='test')`? If it’s the latter, please try adding `dflow.config['namespace']='test'` in the script and then check if there are any remaining issues.

**Comment by heidongxianhua** [comment]
> Could you please provide details on how you change the namespace? Do you change the `config['namespace']` from `dflow.config`, or do you use `Workflow(...,namespace='test')`? If it’s the latter, please try adding `dflow.config['namespace']='test'` in the script and then check if there are any remaining issues.

yes, I change the namespace in config, https://github.com/deepmodeling/dflow/blob/master/src/dflow/config.py#L33C40-L33C40

**Comment by zjgemi** [comment]
Are your Argo components installed in the `test` namespace as well? Because Argo will attempt to look for memoized configmaps in the same namespace by default.

**Comment by heidongxianhua** [comment]
> Are your Argo components installed in the `test` namespace as well? Because Argo will attempt to look for memoized configmaps in the same namespace by default.

yeah, maybe it is. And, which components should I install in the `test` namespace too? @zjgemi

**Comment by zjgemi** [comment]
https://argo-workflows.readthedocs.io/en/latest/managed-namespace/
